Serialize font-style: oblique 14 as oblique

vitorroriz opened the pull request:
#### 6c40de390419f843c064d06d618a4f37869de086
<pre>
Serialize font-style: oblique 14 as oblique
<a href="https://bugs.webkit.org/show_bug.cgi?id=295121">https://bugs.webkit.org/show_bug.cgi?id=295121</a>
<a href="https://rdar.apple.com/154513590">rdar://154513590</a>

Reviewed by NOBODY (OOPS!).

The lack of an &lt;angle&gt; represents 14deg, this also means that 14 deg should be serialized as &apos;oblique&apos; only.
<a href="https://www.w3.org/TR/css-fonts-4/#font-style-prop">https://www.w3.org/TR/css-fonts-4/#font-style-prop</a>

* LayoutTests/fast/text/font-style-parse.html:
* LayoutTests/fast/text/font-style-parse-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-valid-expected.txt:
* Source/WebCore/css/CSSFontStyleWithAngleValue.cpp:
(WebCore::CSSFontStyleWithAngleValue::customCSSText const):
* Source/WebCore/css/CSSFontStyleWithAngleValue.h:
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::identifierForStyleProperty):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6c40de390419f843c064d06d618a4f37869de086

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109537 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29195 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19624 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/114739 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/59740 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/111500 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/29873 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/37783 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83257 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112485 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/23801 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98665 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63717 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23180 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/16808 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59356 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93173 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16848 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/117852 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36576 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27083 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92267 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36949 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94925 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92084 "Found 2 new API test failures: /WebKitGTK/TestWebProcessExtensions:/webkit/WebKitWebView/web-process-crashed, /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37023 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14772 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/32374 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36470 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41941 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36129 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39473 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/37840 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/editing/EditingStyle.cpp:
```diff
@@ -178,6 +178,8 @@ template<typename T> CSSValueID identifierForStyleProperty(T& style, CSSProperty
         auto resolvedAngle = Style::fontStyleAngleFromCSSFontStyleWithAngleValueDeprecated(*fontStyleValue);
         if (!resolvedAngle)
             return CSSValueNormal;
+        else if (resolvedAngle && resolvedAngle == cssFontStyleImplicitObliqueValue)
```
I think line 179 guarantees that `resolvedAngle` can never be `nullptr` here, so I think you can remove that test.

brentfulgham reviewed (approved):
Looks good, assuming no EWS failures. Please consider removing that redundant nullptr check.

vitorroriz committed:
Serialize font-style: oblique 14 as oblique
https://bugs.webkit.org/show_bug.cgi?id=295121
rdar://154513590

Reviewed by NOBODY (OOPS!).

The lack of an <angle> represents 14deg, this also means that 14 deg should be serialized as 'oblique' only.
https://www.w3.org/TR/css-fonts-4/#font-style-prop

* LayoutTests/fast/text/font-style-parse.html:
* LayoutTests/fast/text/font-style-parse-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-valid-expected.txt:
* Source/WebCore/css/CSSFontStyleWithAngleValue.cpp:
(WebCore::CSSFontStyleWithAngleValue::customCSSText const):
* Source/WebCore/css/CSSFontStyleWithAngleValue.h:
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::identifierForStyleProperty):

weinig reviewed (changes requested):
Marking as "Request changes" so I can understand where this stands with the WG first.
